plugins/arrow-flight-rpc/src/main/java/org/opensearch/arrow/flight/stats/FlightStatsResponse.java

```diff
@@ -0,0 +1,194 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.arrow.flight.stats;
+
+import org.opensearch.action.FailedNodeException;
+import org.opensearch.action.support.nodes.BaseNodesResponse;
+import org.opensearch.cluster.ClusterName;
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.common.unit.ByteSizeValue;
+import org.opensearch.core.xcontent.ToXContentObject;
+import org.opensearch.core.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.List;
+
+class FlightStatsResponse extends BaseNodesResponse<FlightNodeStats> implements ToXContentObject {
+
+    public FlightStatsResponse(StreamInput in) throws IOException {
+        super(in);
+    }
+
+    public FlightStatsResponse(ClusterName clusterName, List<FlightNodeStats> nodes, List<FailedNodeException> failures) {
+        super(clusterName, nodes, failures);
+    }
+
+    @Override
+    protected List<FlightNodeStats> readNodesFrom(StreamInput in) throws IOException {
+        return in.readList(FlightNodeStats::new);
+    }
+
+    @Override
+    protected void writeNodesTo(StreamOutput out, List<FlightNodeStats> nodes) throws IOException {
+        out.writeList(nodes);
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        builder.field("cluster_name", getClusterName().value());
+
+        builder.startObject("nodes");
+        for (FlightNodeStats nodeStats : getNodes()) {
+            builder.startObject(nodeStats.getNode().getId());
+            builder.field("name", nodeStats.getNode().getName());
+            builder.field(
+                "streamAddress",
+                nodeStats.getNode().getStreamAddress() != null
+                    ? nodeStats.getNode().getStreamAddress().toString()
+                    : nodeStats.getNode().getAddress().toString()
+            );
+            nodeStats.getMetrics().toXContent(builder, params);
+            builder.endObject();
+        }
+        builder.endObject();
+
+        builder.startObject("cluster_stats");
+        aggregateClusterStats(builder, params);
+        builder.endObject();
+
+        builder.endObject();
+        return builder;
+    }
+
+    private void aggregateClusterStats(XContentBuilder builder, Params params) throws IOException {
+        long totalClientCallsStarted = 0;
```

Comment:
We should look at leveraging the OTel-based framework offered by the `TelemetryAwarePlugin` extension point. I know it's not fully complete yet, but these stat-style APIs are far from ideal for this type of usage.